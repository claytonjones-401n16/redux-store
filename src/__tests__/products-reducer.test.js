import productReducer, {initState} from '../store/productsReducer';

describe('products reducer', () => {
  it('should return the initial state', () => {
    expect(productReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle GET_PRODUCTS', () => {
    expect(
      productReducer(initState, 
        {
          type: 'GET_PRODUCTS',
          payload: [1, 2, 3, 4],
        })
    ).toMatchSnapshot();
  });

  it('should handle ADD_TO_CART', () => {
    expect(
      productReducer([{name: 'test', stock: 10}],
        {
          tyoe: 'ADD_TO_CART',
          payload: 'test'
        })
    ).toMatchSnapshot();
  });

  it('should handle REMOVE_FROM_CART', () => {
    expect(
      productReducer([{name: 'test', stock: 10}],
        {
          tyoe: 'REMOVE_FROM_CART',
          payload: 'test'
        })
    ).toMatchSnapshot();
  })
});