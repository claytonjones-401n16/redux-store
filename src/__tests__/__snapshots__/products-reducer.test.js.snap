// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products reducer should handle ADD_TO_CART 1`] = `
Object {
  "0": Object {
    "name": "test",
    "stock": 10,
  },
}
`;

exports[`products reducer should handle GET_PRODUCTS 1`] = `
Object {
  "addButtonDisabled": false,
  "allProducts": Array [
    1,
    2,
    3,
    4,
  ],
  "productToView": Object {},
}
`;

exports[`products reducer should handle REMOVE_FROM_CART 1`] = `
Object {
  "0": Object {
    "name": "test",
    "stock": 10,
  },
}
`;

exports[`products reducer should return the initial state 1`] = `
Object {
  "addButtonDisabled": false,
  "allProducts": Array [],
  "productToView": Object {},
}
`;
